### PR TITLE
Making blas REQUIRED only when user specified

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -41,45 +41,91 @@ if(BUILD_CAFFE2)
...
-  list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS ${ATLAS_LIBRARIES})
-  list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS cblas)
+# Atlas
+if(BLAS STREQUAL "ATLAS")
+  if(BLAS_SET_BY_USER)
+    find_package(Atlas REQUIRED)
+  else()
+    find_package(Atlas)
+  endif()
+  if(Atlas_FOUND)
```
Comment (yinghai) [inline cmake/Dependencies.cmake]: This can be put into the `else()` branch.

**Review by yinghai — APPROVED** [review]
Overall it makes sense to me. Let's if this fixes the master.

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (pjh5) [reply cmake/Dependencies.cmake]: The include_directories list-appends have to go after both find_package calls though. I could move if(NOT Atlas_FOUND) with a 'break' into the else(), but that doesn't seem worth it to me

File: CMakeLists.txt [hunk 2]
```diff
@@ -145,7 +154,9 @@ endif()
 
 # ---[ Special code path for MKL module
 if (NOT BUILD_ATEN)
-  set(USE_MKL_IDEEP_OR_MKLML ON)
+  if ("${BLAS}" STREQUAL "MKL")
+    set(USE_MKL_IDEEP_OR_MKLML ON)
```
Comment (yinghai) [inline CMakeLists.txt]: Let's remove `USE_MKL_IDEEP_OR_MKML`. I don't think it's used any more.

File: caffe2/mkl/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-if(USE_MKL AND USE_MKLML)
+if(CAFFE2_USE_MKL)
```
Comment (yinghai) [inline caffe2/mkl/CMakeLists.txt]: We can just change it to `USE_MKLML`

File: caffe2/python/CMakeLists.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ set(Caffe2_CPU_PYTHON_SRCS
 )
 
 # ---[ MKL binding
-if(USE_MKL AND USE_MKLML)
+if(CAFFE2_USE_MKL)
```
Comment (yinghai) [inline caffe2/python/CMakeLists.txt]: Same here.

**Comment by yinghai** [comment]
Let's set `USE_IDEEP` and `USE_MKLML` default to `OFF`

**Comment by pjh5** [comment]
Old tests https://ci.pytorch.org/jenkins/job/caffe2-pull-request/5713/

**Comment by pjh5** [comment]
Turned MKLML and IDEEP back on as defaults when MKL is the BLAS in hopes it fixes py2-mkl test error

**Review by orionr — APPROVED** [review]

**Comment by pjh5** [comment]
Old tests https://ci.pytorch.org/jenkins/job/caffe2-pull-request/5733/

**Comment by orionr** [comment]
A version of this landed in https://github.com/pytorch/pytorch/pull/7837

Closing for now - we can reopen later if needed.

**Pull request closed without merging**